sound/soc/sof/intel/cnl.c

```diff
@@ -150,14 +151,53 @@ static void cnl_ipc_dsp_done(struct snd_sof_dev *sdev)
 				CNL_DSP_REG_HIPCCTL_DONE);
 }
 
+static bool cnl_is_ipc_compact(struct snd_sof_ipc_msg *msg)
+{
+	/* the compact IPC subset might be extended later */
+	if (msg->header == (SOF_IPC_GLB_PM_MSG | SOF_IPC_PM_GATE))
+		return true;
+	else
+		return false;
+}
+
+static void cnl_compact_ipc_assemble(struct snd_sof_ipc_msg *msg,
+				     u32 *dr, u32 *dd)
+{
+	struct sof_ipc_pm_gate *pm_gate;
+
+	if (msg->header == (SOF_IPC_GLB_PM_MSG | SOF_IPC_PM_GATE)) {
+		pm_gate = msg->msg_data;
+		/* set cAVS SET_D0IX IPC to the primary register */
+		*dr = CAVS_IPC_MSG_TGT | CAVS_IPC_MOD_SET_D0IX;
+		/* set the extended data register with the flag bits */
+		*dd = pm_gate->flags & CAVS_IPC_MOD_SETD0IX_BIT_MASK;
```

Comment:
so that's where your definitions need to match.

I really don't see why you redefined flags in 'ASoC: SOF: Intel: hda-dsp: add PM_GATE flags definition', and then use a mask that comes from a different header.
You should move the cAVS header earlier, and set those pm_gate->flags using CAVS defines.

Reply:
OK, let me try to remove those from Marcin, and do new implementation to avoid confusion.